libraries/app/database_api.cpp

```diff
@@ -2061,48 +2061,41 @@ vector<blinded_balance_object> database_api_impl::get_blinded_balances( const fl
 //                                                                  //
 //////////////////////////////////////////////////////////////////////
 
-vector<withdraw_permission_object> database_api::get_withdrawals_giver(account_id_type account, uint32_t start, uint32_t limit)const
+vector<withdraw_permission_object> database_api::get_withdrawals_giver(account_id_type account, object_id_type start, uint32_t limit)const
 {
    return my->get_withdrawals_giver( account, start, limit );
 }
 
-vector<withdraw_permission_object> database_api_impl::get_withdrawals_giver(account_id_type account, uint32_t start, uint32_t limit)const
+vector<withdraw_permission_object> database_api_impl::get_withdrawals_giver(account_id_type account, object_id_type start, uint32_t limit)const
 {
    FC_ASSERT( limit <= 100 );
    vector<withdraw_permission_object> result;
 
-   auto withdraw_range = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().equal_range(account);
-   auto total = distance(withdraw_range.first, withdraw_range.second);
-   if(start >= total ) return result;
-   advance(withdraw_range.first, start);
-
-   std::for_each(withdraw_range.first, withdraw_range.second,
-                 [&result] (const withdraw_permission_object& withdraw) {
-                    result.emplace_back(withdraw);
-                 });
-
+   auto withdraw_itr = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>().lower_bound(boost::make_tuple(account, start));
+   while( withdraw_itr->withdraw_from_account == account && result.size() < limit)
```

Comment:
Need to check for `end()`

Reply:
added end of index check at https://github.com/bitshares/bitshares-core/pull/657/commits/845eb374b21bb7963999e1dd96a59952b66e0f1c

for some reason i cant do `_db.get_index_type<withdraw_permission_index>().indices().get<by_from>()` alone. when i do it i get a compiler error as:

```
...
sion_object> >; TagList = boost::mpl::v_item<graphene::chain::by_from, boost::mpl::vector0<mpl_::na>, 0>; Category = boost::multi_index::detail::ordered_unique_tag]’ is protected
   ~ordered_index()
   ^
/home/alfredo/CLionProjects/issue611/libraries/app/database_api.cpp:2074:95: error: within this context
    auto withdraw_idx = _db.get_index_type<withdraw_permission_index>().indices().get<by_from>();
...
```

so i had to code that twice, as far as i remember i never saw the issue in other indexes.